panels/background/cc-background-xml.c

```diff
@@ -434,6 +466,10 @@ cc_background_xml_load_list (CcBackgroundXml *data,
   gchar * datadir;
   gint i;
 
+  datadir = read_vendor_background_directory ();
```

Comment:
Since this can return `NULL`, and `cc_background_xml_load_from_dir()` does not cope with that, I think you should add a `datadir != NULL` check here.